cg/meta/deliver/delivery_api.py

```diff
@@ -147,13 +147,21 @@ def _get_deliverable_samples(self, case: Case, workflow: str) -> list[Sample]:
         deliverable_samples: list[Sample] = []
         samples: list[Sample] = self.store.get_samples_by_case_id(case.internal_id)
         for sample in samples:
-            bundle_name: str = get_bundle_name(case=case, sample=sample, workflow=workflow)
-            bundle_exists: bool = self._bundle_exists(bundle_name=bundle_name, workflow=workflow)
-            sample_is_deliverable: bool = self._is_sample_deliverable(sample)
-            if bundle_exists and sample_is_deliverable:
+            if self.is_sample_deliverable(case=case, sample=sample, workflow=workflow):
                 deliverable_samples.append(sample)
         return deliverable_samples
 
+    def is_sample_deliverable(self, case: Case, sample: Sample, workflow: str) -> bool:
+        bundle_name: str = get_bundle_name(case=case, sample=sample, workflow=workflow)
+        bundle_exists: bool = self._bundle_exists(bundle_name)
+        ignore_missing_bundle: bool = self.ignore_missing_bundle(workflow)
+
+        if not bundle_exists and not ignore_missing_bundle:
+            raise SyntaxError(f"Could not find bundle {bundle_name}")
```

Comment:
If we don't get any files it will raise an error in any case for the sample delivery (`MissingFilesError(f"Could not find any files for sample {sample.internal_id}")`). Not for the case though. That's why the other check is important to keep.

We can look into it later with a more intensive refactoring that this API needs